File: JavaScript/AdapterDesign_pattern.js
class IPaymentProcessor {
    pay() {
        throw new Error("Not implemented");
    }
}

class Application extends IPaymentProcessor {
    pay() {
        console.log("Legacy code");
    }
}

class StripePayment {
    makePayment() {
        console.log("Stripe is making payment");
    }
}

class StripeAdapter extends IPaymentProcessor {
    constructor(stripePayment) {
        super();
        this.stripePayment = stripePayment;
    }

    pay() {
        this.stripePayment.makePayment();
    }
}

class PayTM {
    makePaymentPaytm() {
        console.log("paytm karo");
    }
}

class PayTMAdapter extends IPaymentProcessor {
    constructor(paytm) {
        super();
        this.paytm = paytm;
    }

    pay() {
        this.paytm.makePaymentPaytm();
    }
}

function clientCode(process) {
    process.pay();
}

class AdapterDesign_pattern {
    static main(args) {
        const process = new StripeAdapter(new StripePayment());

        process.pay();
    }
}

AdapterDesign_pattern.main([]);

File: Java/AdapterDesign_pattern.java

interface IPaymentProcessor{
    void pay();
}

class Application implements IPaymentProcessor{
    @Override
    public void pay() {
        System.out.println("Legacy code");
    }
}

class StripeAdapter implements IPaymentProcessor{
    StripePayment stripePayment;
    @Override
    public void pay() {
        stripePayment.makePayment();
    }
}

class StripePayment{
    void makePayment(){
        System.out.println("Stripe is making payment");
    }
}

class PayTM{
    void makePaymentPaytm(){
        System.out.println("paytm karo");
    }
}

class PayTMAdapter implements IPaymentProcessor{
    PayTM paytm;

    @Override
    public void pay() {
        paytm.makePaymentPaytm();
    }

}

// class UniversalAdapter implements IPaymentProcessor{
//     PayTM payTM;
//     StripePayment stripe;
//     //sadsa
//     //dasda
//     //adsasddads
//     //dasdasdasd
//     //dasdasdada
//     //dsadasdasd

// }


public class AdapterDesign_pattern {
    public void clientCode(IPaymentProcessor process){
        process.pay();
    }
    public static void main(String[] args) {
        IPaymentProcessor process = new StripeAdapter();

        //--------------------
        process.pay();
    }
}
